llvm/include/llvm/Transforms/Coroutines/CoroSplit.h

```diff
@@ -21,19 +21,26 @@
 
 namespace llvm {
 
+namespace coro {
+class BaseABI;
+class Shape;
+} // namespace coro
+
 struct CoroSplitPass : PassInfoMixin<CoroSplitPass> {
-  const std::function<bool(Instruction &)> MaterializableCallback;
+  // BaseABITy generates an instance of a coro ABI.
+  using BaseABITy = std::function<coro::BaseABI *(Function &, coro::Shape &)>;
```

Comment:
Let's move the definition and its use closer. And also `BaseABITy` may not be good name. If it is only used once, maybe it is not to not give it a name right now.

Reply:
Done